umake/frameworks/scala.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2014 Canonical
+#
+# Authors:
+#  Didier Roche
+#  Igor Vuk
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation; version 3.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+
+"""Scala module"""
+
+from contextlib import suppress
+from gettext import gettext as _
+import logging
+import os
+import re
+import umake.frameworks.baseinstaller
+from umake.interactions import DisplayMessage
+from umake.tools import add_env_to_user
+from umake.ui import UI
+
+logger = logging.getLogger(__name__)
+
+
+class ScalaCategory(umake.frameworks.BaseCategory):
+
+    def __init__(self):
+        super().__init__(name="Scala", description=_("The Scala Programming Language"), logo_path=None)
+
+
+class ScalaLang(umake.frameworks.baseinstaller.BaseInstaller):
+
+    def __init__(self, category):
+        super().__init__(name="Scala Programming Language", description=_("Scala compiler and interpreter (default)"),
```

Comment:
A small change would be nice for: name="Scala Programming Language"
-> this will indeed translate as a command to:
"umake scala scala-programming-language"

This is quite long to type (even if you set this framework as the default for its category, I would rather see a shorter command).